builtin/logical/database/backend.go

```diff
@@ -140,11 +155,24 @@ func (b *databaseBackend) Role(s logical.Storage, roleName string) (*roleEntry,
 		return nil, nil
 	}
 
+	var upgradeCh upgradeCheck
+	if err := entry.DecodeJSON(&upgradeCh); err != nil {
+		return nil, err
+	}
+
 	var result roleEntry
 	if err := entry.DecodeJSON(&result); err != nil {
 		return nil, err
 	}
 
+	empty := upgradeCheck{}
+	if upgradeCh != empty {
```

Comment:
I believe if there are any matching fields without the typo (revocation, renew, rollback), this will evaluate as true, which could overwrite the creation statements with blank.

Reply:
I think this is an all-or-nothing decode since the top level `statments` was misspelled.

Reply:
Yeah, I misread it as decoding the nested object.  All good here.